x-pack/plugins/lists/server/services/exception_lists/exception_list_client.ts

```diff
@@ -66,15 +74,52 @@ import {
   importExceptionsAsArray,
   importExceptionsAsStream,
 } from './import_exception_list_and_items';
+import { DataValidationError } from './utils/errors';
+import {
+  transformCreateExceptionListItemOptionsToCreateExceptionListItemSchema,
+  transformUpdateExceptionListItemOptionsToUpdateExceptionListItemSchema,
+} from './utils';
 
 export class ExceptionListClient {
   private readonly user: string;
-
   private readonly savedObjectsClient: SavedObjectsClientContract;
-
-  constructor({ user, savedObjectsClient }: ConstructorOptions) {
+  private readonly serverExtensionsClient: ExtensionPointStorageClientInterface;
+  private readonly disableServerExtensionPoints: boolean;
+
+  constructor({
+    user,
+    savedObjectsClient,
+    serverExtensionsClient,
+    disableServerExtensionPoints = false,
+  }: ConstructorOptions) {
     this.user = user;
     this.savedObjectsClient = savedObjectsClient;
+    this.serverExtensionsClient = serverExtensionsClient;
+    this.disableServerExtensionPoints = disableServerExtensionPoints;
+  }
+
+  /**
+   * Validates some data with a given validator (`io-ts` schema) and if validation errors exist, then an
+   * Error instance will be returned.
+   *
+   * @param data
+   * @param validator A `io-ts` codec that will be used to `.decode()` the data and then check it for validation errors
+   *
+   * @private
+   */
+  private validateData<D>(validator: t.Type<D>, data: D): undefined | DataValidationError {
+    return pipe(
+      validator.decode(data),
+      (decoded) => exactCheck(data, decoded),
+      fold(
+        (errors: t.Errors) => {
+          const errorStrings = formatErrors(errors);
+
+          return new DataValidationError(errorStrings, 400);
+        },
+        () => undefined
+      )
+    );
```

Comment:
For the `validateData` function here since it's pure can we pull this out of the class completely and move it into its own file named something like `validate_data.ts` and put that file somewhere such as in this folder?

```sh
x-pack/plugins/lists/server/services/exception_lists/utils
```

Within this class you should notice that we try to immediately go from class/method to a pure function as soon as we can. The reasoning, is that for the most part we try to use pure functions and not very many classes. This shouldn't take very long, but would help us out a lot to keep this class from accumulating private methods, and functionality and only getting bigger where it's needed to grow. We really want there to be very few tests with this class and keep this class fairly "dumb" and "simple" where possible. It should be easy enough to skim this file and see everything is passing off to something else basically.

fwiw, testing and using pure functions prevents a lot of problems we have seen in the past which we promote heavily:
* pure functions can be re-used and called from anywhere as long as the caller has the deps needed to pass in. Class methods and stateful functions cannot easily.
* pure functions are incredibly simple to see if they have unit tests written when they're standalone files and if they're not standalone files, they're easy enough to break into individual standalone files.
* Writing unit tests for pure functions is typically as simple as a class if not simpler as you're just setting up deps and passing them in. 
* pure functions are easier to move around without side effects and state issues between folders and projects including if we want to move them into a package.
* It's easier to remove and simply imports with pure functions and see relationships than large classes and private/public methods.